### DISCUSSION TITLE
Duplicate name records in DNS and /etc/hosts

### DISCUSSION THREAD

**Discussion by sgaosdgr** [bluebanquise/bluebanquise#874] [Cluster deployment]
Hi, 

After applying role dns_server and hosts_file, a duplicate name/dns record is created for each node.

For example: 
```
192.168.1.251 compute-0-5-net-1 
192.168.1.251 compute-0-5 compute-0-5.cluster.local
```

Tried changing `hosts_file_range`, but it does not make any difference. 

Any idea how to prevent a duplicate record ending with net-1 from being created?

**Comment by oxedions** [comment]
Dear @sgaosdgr 

Yes, I agree this should be improved. The introduction of entries for all networks a node is attached to was added to help advanced networking, with cluster that wanted to use for example Ethernet to manage some services, and Infiniband for other services, etc. They could bind to the desired network using this.

There is I think 2 things to improve here:

1. Make this optional, as not everyone need this feature, so only direct hostname exists (this is not compatible with the ssh_client role, but most users do not need it)
2. If used, ensure identical records are merged at least in /etc/hosts (this is not possible in DNS forward file).

Would that be what you are looking for ? :blush:

**Comment by sgaosdgr** [comment]
If dual host name entries are needed by a role, can we make the real host name entry as first?  For example:

```
192.168.1.251 compute-0-5 compute-0-5.cluster.local
192.168.1.251 compute-0-5-net-1
```

This way, the real host name is always used, instead of a host name alias. 

Providing an option to generate the network based host name is good. 

In my case, reversing the host name entry order should solve the issue.

**Comment by oxedions** [comment]
Dear @sgaosdgr 

Ok, this is simple to fix. I will do that asap :blush:

**Comment by sgaosdgr** [comment]
Thank you.

**Comment by oxedions** [comment]
Ok now it should be good :slightly_smiling_face:  https://github.com/bluebanquise/bluebanquise/pull/878
